pandas/_libs/lib.pyx

```diff
@@ -865,8 +865,8 @@ def is_all_arraylike(obj: list) -> bool:
 
     for i in range(n):
         val = obj[i]
-        if not (isinstance(val, list) or
-                util.is_array(val) or hasattr(val, "_data")):
+        if not (isinstance(val, list) or util.is_array(val)
+                or hasattr(val, "_mgr") or hasattr(val, "_data")):
```

Comment:
Wouldn't this now include DataFrames? I think that's a little too broad. Maybe just scope it to isinstance(..., Series)?

@jbrockmendel probably has a better idea

Reply:
yah the circular import sucks but the right way to do this check is with an isinstance for the thing(s) we actually want to catch here

Reply:
> Wouldn't this now include DataFrames? I think that's a little too broad. Maybe just scope it to isinstance(..., Series)?

I think DataFrame also has the `_data` attribute, so at least this is not breaking the original behavior (if there are nothing other than NDFrame that has the `_mgr` attribute).

As @jbrockmendel has said, we cannot explicitly import those classes here. Unless... we simply use `all(...)` in Python instead of this Cython function?

Reply:
if the perf hit isnt too bad doing it outside cython works. or could runtime-import ABCSeries and ABCIndex